tests/test_magnetostatics.py

```diff
@@ -0,0 +1,183 @@
+):
+    """Compute pairwise distance vectors."""
+    atom_is = neighbor_indices[:, 0]
+    atom_js = neighbor_indices[:, 1]
+
+    pos_is = positions[atom_is]
+    pos_js = positions[atom_js]
+
+    distance_vectors = pos_js - pos_is
+
+    if cell is not None and neighbor_shifts is not None:
+        shifts = neighbor_shifts.type(cell.dtype)
+        distance_vectors += shifts @ cell
+    elif cell is not None and neighbor_shifts is None:
+        raise ValueError("Provided `cell` but no `neighbor_shifts`.")
+    elif cell is None and neighbor_shifts is not None:
+        raise ValueError("Provided `neighbor_shifts` but no `cell`.")
+
+    return distance_vectors
+
+
+class System:
+    def __init__(self):
+        self.cell = torch.tensor(
+            [[10.0, 0.0, 0.0], [0.0, 10.0, 0.0], [0.0, 0.0, 10.0]], dtype=torch.float64
+        )
+        self.dipoles = torch.tensor(
+            [[1.0, 1.0, 0.0], [1.0, 1.0, 0.0], [1.0, 1.0, 0.0]], dtype=torch.float64
+        )
+        self.positions = torch.tensor(
+            [[0.0, 0.0, 0.0], [0.0, 2.0, 0.0], [0.0, 4.0, 0.0]], dtype=torch.float64
+        )
+        self.neighbor_indices = torch.tensor(
+            [[0, 1], [1, 2], [0, 2]], dtype=torch.int64
+        )
+        self.neighbor_vectors = torch.tensor(
+            [[0.0, 2.0, 0.0], [0.0, 2.0, 0.0], [0.0, 4.0, 0.0]], dtype=torch.float64
+        )
+
+
+system = System()
+
+
+def test_magnetostatics_direct():
+    calculator = CalculatorDipole(
+        potential=PotentialDipole(dtype=torch.float64),
+        full_neighbor_list=False,
+        dtype=torch.float64,
+    )
+    pot = calculator(
+        dipoles=system.dipoles,
+        cell=system.cell,
+        positions=system.positions,
+        neighbor_indices=system.neighbor_indices,
+        neighbor_vectors=system.neighbor_vectors,
+    )
+    result = torch.einsum("ij,ij->i", pot, system.dipoles).sum()
+    expected_result = torch.tensor(-0.2656, dtype=torch.float64)
+    assert torch.isclose(result, expected_result, atol=1e-4), (
+        f"Expected {expected_result}, but got {result}"
+    )
+
+
+@pytest.mark.parametrize(
+    ("smearing", "sr_potential"),
+    [
+        (1e10, torch.tensor(-0.2656, dtype=torch.float64)),
+        (1e-10, torch.tensor(0.0000, dtype=torch.float64)),
+    ],
+)
+def test_magnetostatics_sr(smearing, sr_potential):
+    calculator = CalculatorDipole(
+        potential=PotentialDipole(smearing=smearing, dtype=torch.float64),
+        full_neighbor_list=False,
+        lr_wavelength=1.0,
+        dtype=torch.float64,
+    )
+    pot = calculator._compute_rspace(
+        dipoles=system.dipoles,
+        neighbor_indices=system.neighbor_indices,
+        neighbor_vectors=system.neighbor_vectors,
+    )
+    result = torch.einsum("ij,ij->i", pot, system.dipoles).sum()
+    expected_result = sr_potential
+    assert torch.isclose(result, expected_result, atol=1e-4), (
+        f"Expected {expected_result}, but got {result}"
+    )
```

Comment:
You don't have to put an error message here, just use `torch.testing.assert_close`